types/wlr_layer_shell_v1.c

```diff
@@ -498,7 +509,7 @@ struct wlr_layer_shell_v1 *wlr_layer_shell_v1_create(struct wl_display *display)
 	}
 
 	struct wl_global *global = wl_global_create(display,
-		&zwlr_layer_shell_v1_interface, 2, layer_shell, layer_shell_bind);
+		&zwlr_layer_shell_v1_interface, zwlr_layer_shell_v1_interface.version, layer_shell, layer_shell_bind);
```

Comment:
We can't advertise `zwlr_layer_shell_v1_interface.version` here, because re-compiling with an XML file that has v5 won't make the wlroots code handle v5 properly.

Reply:
Ok, I just rebased on master after #2608 -- I'm still wondering if it would make sense to allow a compositor to request a lower version than available in wlroots (so if it does not support the new version, only the lower version will be advertised to clients). It would still be a breaking change though since that would need to be an extra parameter for this function.

Reply:
Our policy so far has been that compositors always need to support the version supported by wlroots.

In this particular case it's not a big deal because compositors are not required to focus layer surfaces.